Esan Miracle Olaoluwa/Faizal's code Update Function/composition.ts

```diff
@@ -0,0 +1,132 @@
+class Laptop {
+    keyboard: Keyboard[] = [];
+    nic: NetworkInterfaceCard;
+    screen: Display;
+    hardDisk: HardDisk;
+    powerState: boolean = false;
+    bitKind: BitKind;
+    os: OS;
+    allUpdateLogs: string[] = [];
+  
+    constructor(
+      keyboard: Keyboard,
+      bit: BitKind,
+      display: Display,
+      nic: NetworkInterfaceCard,
+      os: OS,
+      harddisk: HardDisk
+    ) {
+      this.keyboard.push(keyboard);
+      this.bitKind = bit;
+      this.screen = display;
+      this.nic = nic;
+      this.os = os;
+      this.hardDisk = harddisk;
+    }
+  
+    switchOn() {
+      this.powerState = true;
+    }
+  
+    switchOff() {
+      this.powerState = false;
+    }
+  
+    showAllUpdateLogs() {
+      console.log("All updates made: " + this.allUpdateLogs.join("\n"));
+    }
+  
+    update(thingToUpdate: string, newValue: any) {

```

Comment:
While this will solve the problem, it does not solve the problem using generics.

Ask 'Elochuwkwu' in the group to put you through.
Also check youtube and articles for 'Generics' 

* Generic classes
* Generic functions